internal/appconfig/validation.go

```diff
@@ -18,7 +19,7 @@ func (cfg *Config) Validate(ctx context.Context) (err error, extra_info string)
 		return errors.New("App config file not found"), ""
 	}
 
-	platformVersion := NomadPlatform
+	platformVersion := cfg.platformVersion
```

Comment:
here be 🐉 but we can ⚔️ them

The only reason we call `GetAppBasic()` below is to get the platform version when it is unknown, but if it is already known we shouldn't be making an extra roundtrip to Web API.

But when it is unknown, meaning `cfg.PlatformVersion == nil`, we have to default to NomadPlatform to be backwards compatible in case the app doesn't exists.

Reply:
updated